packages/opensearch/src/fixIds.ts

```diff
@@ -133,9 +133,30 @@ const fixId = (old: any) => {
 
 const fixIdValue = (val: string) => (val === 'id' ? '_id' : val)
 
-// biome-ignore lint/suspicious/noExplicitAny: <explanation>
-const clean = (obj: any) =>
+// Remove unnecessary undefined properties
+type JSObj = Record<string, unknown>
+const clean = <T extends JSObj>(obj: T): T =>
```

Comment:
We could also use `structuredClone(obj)`, but it's a bit slower since V8 [has optimized `JSON.parse(JSON.stringify())`](https://willcodefor.beer/deep)

Reply:
All I want to do is remove undefineds... so please replace :)